### PR TITLE
Change default flash behavior for `viewer` screenshot-like methods (GUI functionality remains the same)

### PR DESCRIPTION
**Pull request by TimMonko**
# References and relevant issues

- Addresses one task in #7433, by changing defualt flash behavior in `viewer` methods. This was discussed in #7427.
- Effectively reverts changes in #7427, because they are no longer required to prevent issues brought up in #7425

# Description

- Changes default from `flash=True` to `flash=False` in `viewer.screenshot()` and `viewer.export_figure()` methods. 
- Adds test to assert that `viewer.screenshot()` does not flash by default, and that flashing argument does trigger successfully. This appears to work, but the '_flash_animation' attribute is only temporary. Is there a test environment where this theoretically wouldn't work? The flash animation appears to be slow (i.e. probably the root of the problem in #7425). 
- Maintains current GUI functionality because non-`viewer` methods are used. Flash animation continues when triggered via GUI methods and shortcuts
- Reverts `flash=False` additions in #7427 in relevant examples because it is no longer required with this change. 

This PR changes the default flash animation behavior for `viewer` screenshot-like methods because this is the presented way for users to take screenshots via scripting. By changing flash animation to default, this PR prevents naively triggering rapid flashing from scripting or a plugin that uses `viewer` methods to save a view of the viewer/canvas. This is a somewhat opinionated change (because a user could already access and change this argument), but 1) increases accessibility for naive users by not triggering a flash and 2) scripting explicitly calls the viewer methods, so users do not need the flash animation for feedback [where, in the GUI, this is helpful, and remains].

This PR does not change the default flash behavior in `qt_viewer.py:QtViewer` and `qt_main_window.py:Window` because `viewer` methods are built on top. Should these other methods be changed? They look to be public API in one sense (i.e. `viewer.window.export_figure`) but also considered private API according to the architecture since they are in `_qt/`. If these other methods are changed, then `_qt/_qapp_model/_file.py` would also need updated for Action callbacks. 

ps. Despite this being a small PR, it took me a while to navigate and tease apart these systems, so I'm grateful to have learned more about the bones of napari! If anyone cares to answer, why are there these 'layers' to the framework? It could be structured differently so that there would not be so much repetitive code across different files, and non-'identical' methods to call the same functionality. What is the advantage of this layer of the API? Does it make certain things possible that otherwise wouldn't be?

### CODE REVIEW

**Comment by DragaDoncila** [comment]
@TimMonko thanks for opening this one! @Czaki is probably the best person for code review here, although maybe @psobolewskiPhD you'd like to play with this and make sure it works as intended. 

> If anyone cares to answer, why are there these 'layers' to the framework? It could be structured differently so that there would not be so much repetitive code across different files, and non-'identical' methods to call the same functionality. What is the advantage of this layer of the API?

Great questions! I'm definitely not the best person to answer this, probably @Czaki is the man for the job, but would you mind expanding a little on which layers you are referring to here? Is it the relationship between `viewer`, `Window` and `QtMainWindow`, or is it something else?

**Comment by TimMonko** [comment]
> Is it the relationship between `viewer`, `Window` and `QtMainWindow`, or is it something else

Yes, that's a big part of the question. I imagine some of it has to do with trying to separate out the Qt backend as a requirement. I think the other piece though is why almost replicate the method in each of these layers, rather than have one main method which is called from every instance, instead of basically the choice between each version.

**Comment by Czaki** [comment]
> @TimMonko thanks for opening this one! @Czaki is probably the best person for code review here, although maybe @psobolewskiPhD you'd like to play with this and make sure it works as intended.

The code is ok. But I do not have a strong opinion about changing the behavior.

**Comment by jni** [comment]
> ps. Despite this being a small PR, it took me a while to navigate and tease apart these systems, so I'm grateful to have learned more about the bones of napari! If anyone cares to answer, why are there these 'layers' to the framework? It could be structured differently so that there would not be so much repetitive code across different files, and non-'identical' methods to call the same functionality. What is the advantage of this layer of the API? Does it make certain things possible that otherwise wouldn't be?

The answer is "it depends" 😅. One goal, which we've only been partially successful at, is to separate the napari application/data model (viewers, layers, layer list, camera, etc) from the *renderer* (Vispy) and the UI (Qt), so that in the future we can use different backends to actually display data. This goal is explained in [this document about our architecture](https://napari.org/stable/developers/architecture/napari_models.html#napari-model-event). (You can see other explanatory documents in [this page](https://napari.org/stable/developers/architecture/index.html), with special thanks to @lucyleeow.)

But, in many cases the answer is, the layers have accumulated in the same way that sedimentary layers accumulate on the Earth, or layers of paint on a house — because development and fixes happen in pieces from a whole bunch of contributors (like yourself! 🙏 😊), sometimes the easiest way to do something is to add a layer. It takes substantial effort to peel them back and simplify, and we haven't had the chance to do it, though we intend to make significant efforts in this direction in the coming years.

In short, if you see something specific that you think could be simpler, please ping us (issue or Zulip or community meeting) and we can discuss if it makes sense or if it's a "load bearing layer", so to speak. 😂 We are definitely open to contributions that simplify code!

**Review by jni — APPROVED** [review]

**Review by Czaki — APPROVED** [review]

**Comment by TimMonko** [comment]
> But, in many cases the answer is, the layers have accumulated in the same way that sedimentary layers accumulate on the Earth, or layers of paint on a house — because development and fixes happen in pieces from a whole bunch of contributors (like yourself! 🙏 😊), sometimes the easiest way to do something is to add a layer. It takes substantial effort to peel them back and simplify, and we haven't had the chance to do it, though we intend to make significant efforts in this direction in the coming years.

Thanks @jni, this is a helpful perspective to understand! I'll keep plugging along with things and keep trying to contribute too! Appreciate the time you took to reply :)

**Comment by brisvag** [comment]
Follow-up proposal: since the point of flashing was to show the user that *something* happened, could we introduce a notification to serve the same purpose?

**Comment by jni** [comment]
I don't feel like this is necessary when running from a script. (The logger would be a better place for this @brisvag ;)

**Comment by jni** [comment]
Wait, are you saying that you are using the GUI and not seeing the flashing? Because that was the opposite of the intent of this PR! (As @TimMonko has emphasised with his posthumous title update 😂)

**Comment by psobolewskiPhD** [comment]
I get a flash with current main when using the menu items.

**Comment by TimMonko** [comment]
@brisvag clarified in the community meeting he did not test it and was just reading through. That's why I updated the title to reflect the functionality 🙃
